js/hw9_2.js

```diff
@@ -0,0 +1,83 @@
+// task2
+const OPERATIONS = ['+', '-', '/', '*'];
+let newRound = true;
+
+let result = $(".calculator__result")[0];
+
+$(".calculator__digit").on('click', function () {
+
+    if (newRound === true) { result.innerText = ''; newRound = false };
+
+    // this can be written with less code, but here readability is more important. 
+    // if previous symbol was a number
+    if (OPERATIONS.indexOf(result.innerText[result.innerText.length - 1]) == -1) {
+        result.innerText += this.innerText;
+    }
+    else {
+        // if previous symbol was an operation, but it's negative sign and there is one more operator before
+        if (OPERATIONS.indexOf(result.innerText[result.innerText.length - 3]) > -1 &&
+            result.innerText[result.innerText.length - 1] == '-') {
+            result.innerText += this.innerText;
+        } else
+            // if it's first negative number
+            if (result.innerText.length == 1) {
+                result.innerText += this.innerText;
+
+            }
+            else {
+                result.innerText += ' ' + this.innerText;
+            }
+    }
+})
+
+$(".calculator__operation").on('click', function () {
+    if (newRound === true) { result.innerText = ''; newRound = false };
+
+    result.innerText += ' ' + this.innerText;
+})
+
+$(".calculator__clear").on('click', function () {
+    result.innerText = '';
+})
+
+$(".calculator__equal").on('click', function () {
+    let input = result.innerText.split(' ');
+
+    if (input.length < 3) {
+        result.innerText += ' = bad request';
```

Comment:
It's good. But in real calculators - we cannot show and see text. You can simply just do not calculate if something is wrong

Reply:
And this kind of checking is kind of unsafe. You need to check each value, not the final expression with `.length `property